src/v0/destinations/am/transform.js

```diff
@@ -142,6 +144,9 @@ function prepareTraitsConfig(configPropertyTrait, actionKey, traitsObject) {
     const property = traitsElement.traits;
     traitsObject = updateTraitsObject(property, traitsObject, actionKey);
   });
+  if(Object.keys(traitsObject[actionKey]).length === 0) {
+    delete traitsObject[actionKey];
+  }
```

Comment:
Rather than doing this, store all the updated traits in a temporary object and assign at the end if it is valid.

Reply:
We are using `traitsObject` in the `updateTraitsObject` function and manipulating `traitsObject` due which we are doing this.